fn/goroutine_manager.go

```diff
@@ -29,8 +25,9 @@ func NewGoroutineManager(ctx context.Context) *GoroutineManager {
 	}
 }
 
-// Go starts a new goroutine if the manager is not stopping.
-func (g *GoroutineManager) Go(f func(ctx context.Context)) error {
+// Go starts a new goroutine if the manager is not stopping and returns if it
+// started a new goroutine.
```

Comment:
Nit: Go tries to start a new goroutine and returns a boolean indicating its success ?

Reply:
Fixed. I added a note about when it succeeds.

```go
// Go tries to start a new goroutine and returns a boolean indicating its
// success. It fails iff the goroutine manager is stopping or its context passed
// to NewGoroutineManager has expired.
func (g *GoroutineManager) Go(f func(ctx context.Context)) bool {
```